instrumentation/opentelemetry_cowboy/src/opentelemetry_cowboy.erl

```diff
@@ -57,16 +57,22 @@ handle_event([cowboy, request, stop], Measurements, Meta, _Config) ->
                   {'http.response_content_length', maps:get(resp_body_length, Measurements)}
                  ],
     otel_span:set_attributes(Ctx, Attributes),
-    case Status of
+    Status1 = transform_status(Status),
+    case Status1 of
         undefined ->
-            {ErrorType, Error, Reason} = maps:get(error, Meta),
-            otel_span:add_event(Ctx, atom_to_binary(ErrorType, utf8), [{error, Error}, {reason, Reason}]),
-            otel_span:set_status(Ctx, opentelemetry:status(?OTEL_STATUS_ERROR, Reason));
-        Status when Status >= 400 ->
-            otel_span:set_attributes(Ctx, [{'http.status', Status}]),
+          case maps:get(error, Meta, undefined) of
+            {ErrorType, Error, Reason} ->
+              otel_span:add_event(Ctx, atom_to_binary(ErrorType, utf8), [{error, Error}, {reason, Reason}]),
+              otel_span:set_status(Ctx, opentelemetry:status(?OTEL_STATUS_ERROR, Reason));
+            _ ->
+              % do nothing first as I'm unsure how should we handle this
+              ok
+          end;
```

Comment:
I'm not sure how would we want to handle the other scenario where we might have `undefined` status code but is not an error. As far as I know, it could be a websocket connection based on the stacktrace below:

```
Reason={:badkey, :error}
Stacktrace=[
  {:erlang, :map_get,
   [
     :error,
     %{
       pid: #PID<0.708.0>,
       ref: TesteWeb.Endpoint.HTTP,
       req: %{
         body_length: 0,
         cert: :undefined,
         has_body: false,
         headers: %{
           "accept" => "*/*",
           "accept-encoding" => "gzip, deflate",
           "accept-language" => "en-US,en;q=0.5",
           "cache-control" => "no-cache",
           "connection" => "keep-alive, Upgrade",
           "cookie" => "user_data=eyJoZXhfY29sb3IiOiIjNjBBNUZBIiwibmFtZSI6bnVsbH0=; _teste_key=SFMyNTY.g3QAAAABbQAAAAtfY3NyZl90b2tlbm0AAAAYTnM1Q254VGhPMWRva2hBem1zOFJKcmVO.-cfAkiIzqtt3BPMkQYFfNZWOd-iijLzaNEz1CAsly0o",
           "host" => "localhost:4000",
           "origin" => "http://localhost:4000",
           "pragma" => "no-cache",
           "sec-fetch-dest" => "websocket",
           "sec-fetch-mode" => "websocket",
           "sec-fetch-site" => "same-origin",
           "sec-websocket-extensions" => "permessage-deflate",
           "sec-websocket-key" => "MUMOmAyxUWbViEGZl/pJvA==",
           "sec-websocket-version" => "13",
           "upgrade" => "websocket",
           "user-agent" => "Mozilla/5.0 (Macintosh; Intel Mac OS X 10.15; rv:94.0) Gecko/20100101 Firefox/94.0"
         },
         host: "localhost",
         method: "GET",
         path: "/phoenix/live_reload/socket/websocket",
         peer: {{127, 0, 0, 1}, 53051},
         pid: #PID<0.708.0>,
         port: 4000,
         qs: "vsn=2.0.0",
         ref: TesteWeb.Endpoint.HTTP,
         scheme: "http",
         sock: {{127, 0, 0, 1}, 4000},
         streamid: 1,
         version: :"HTTP/1.1"
       },
       resp_headers: :undefined,
       resp_status: :undefined,
       streamid: 1
     }
   ], [error_info: %{module: :erl_erts_errors}]},
```

Reply:
I'm not sure what to do, either. I would have thought a websocket upgrade request would return `101` on `resp_status` if this stacktrace was triggered in a stop event. I'm not well-versed in how cowboy manages that sequence.